### PR TITLE
More metrics for fs cache

### PR DESCRIPTION
**Pull request by kssenii**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-clickhouse** [comment]
*This is an automated comment for commit 17dda0d8c34b009474c0e3ac8401080e86d54061 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/56165/17dda0d8c34b009474c0e3ac8401080e86d54061/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/6705955907">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/56165/17dda0d8c34b009474c0e3ac8401080e86d54061/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: src/Interpreters/Cache/FileSegment.cpp [hunk 1]
```diff
@@ -917,10 +924,11 @@ void FileSegment::use()
     }
 }
 
-FileSegments::iterator FileSegmentsHolder::completeAndPopFrontImpl()
+FileSegmentsHolder::FileSegmentsHolder(FileSegments && file_segments_, bool complete_on_dtor_)
+    : file_segments(std::move(file_segments_)), complete_on_dtor(complete_on_dtor_)
 {
-    front().complete();
-    return file_segments.erase(file_segments.begin());
+    CurrentMetrics::add(CurrentMetrics::FilesystemCacheHoldFileSegments, file_segments.size());
```
Comment (alesapin) [inline src/Interpreters/Cache/FileSegment.cpp:930]: Why do we need both profile event and metric?

**Review by alesapin — APPROVED** [review]

File: src/Interpreters/Cache/FileSegment.cpp (hunk 1 above)
Comment (kssenii) [reply src/Interpreters/Cache/FileSegment.cpp:930]: Because they serve different purposes

**Comment by kssenii** [comment]
> Stateless tests (debug) [3/5] — fail: 1

01414_mutations_and_errors_zookeeper

**Pull request merged**